src/dataStructures.ts

```diff
@@ -0,0 +1,74 @@
+/* -----------------------------------------------------------------------
+                    Virtual Field Environment Software
+   -----------------------------------------------------------------------
+    Instantiation of the objects that will be used for interacting with 
+    Photo Sphere Viewer to build a virtual geology field guide.
+   ----------------------------------------------------------------------- */
+
+// Virtual Field Environment: the total collection of photo sphere environments
+export interface VFE {
+  name: string;
+  map: Map<string, number>; // arguments <"imgURL", center_number>
+  photospere: photosphere[];
+}
+
+// Navigation map: a birdseye view of the various hotspots within a single 360-environment
+export interface navMap {
+  src: string;
+  hotspot: hotspotData[];
+}
+
+// Photosphere: a single 360-environment
+export interface photosphere {
+  id: string;
+  src: string;
+  hotspot: hotspotData[];
+  backgroundAudio: string;
+}
+
+// hotSpotData: types of media resources for a hotspot within a photosphere
+export type hotspotData = img | audio | video | URL | doc | photosphereLink;
+
+// media objects
+export interface img {
+  tag: "img";
+  src: string;
+  hotspot: img2d | img3d; // can be 2d|3d
```

Comment:
A 2D image can only have 2D hotspots on it, so this field should be `hotspots: Hotspot2D[]`.